handbook/engineering/developer-insights/code-insights/processes.md

```diff
@@ -16,28 +16,80 @@ Attendees are encouraged to add comments on talking points asynchronously before
+
+### Releases
+
+Despite following two-week iterations, our [releases are monthly on the 20th](../../releases.md#releases) and we may sometimes need to order tasks in a way that we can get important projects into the next release.
+
+We also intentionally plan so we can avoid merging significant work less than two days before a release (if a release is on the 20th, our last day to merge is the 18th). Exceptions require explicit approval of both the PM and EM.
+
+### Planning larger projects
+
+Larger (usually multi-iteration) projects coming from our roadmap go through three phases:
+
+1. **Product context and scope agreement** (usually 1-2 weeks)<br>
+   The PM creates a product-level RFC summarizing user desires from feedback and proposes a reasonable timeline to reach a scope agreement (mentioned in the RFC).
+   Engineers, designer (as applicable), PM, and EM discuss the needs in the RFC and the feasibility/complexity of different options.
+   **The goal of this phase is to agree on a scope for the project**, which will generally be a _subset_ of the desires the PM outlined in the RFC based on the implementation complexity, how long it would roughly take to implement, and how much time we want to dedicate to the project.
+   After discussion, the agreed-upon scope is recorded in a section at the bottom of the RFC.
+
+2. **Design and implementation planning** (usually 1-2 weeks)<br>
+   For a UI project, our designer will create wireframes and designs which are reviewed and discussed by all teammates in Figma.
+   Similarly, engineers will make a proposal for the needed GraphQL APIs (e.g. in an RFC) and agree on the API shape, think the implementation through, and create concrete implementation sub-tasks with estimates, testing plans, and a tracking issue collecting them.
+   The goal of this phase is to be "ready to go" for the implementation phase – having a plan (with reasonable confidence) that only needs to be executed.
+
+3. **Implementation and testing** (usually 1-4 weeks)<br>
+   Engineers execute on the implementation plan, putting a set of issues from the tracking issue into each iteration.
+   This also includes that each sub-implementation-task is sufficiently tested, meaning by the end of this phase the project is ready to ship to customers with confidence.
+
+We sequentialize and parallelize projects in a way that we can _plan_ projects (step 1 and 2) while another project is currently being _implemented_, so that we always have the next project ready to be implemented by the time a project has finished implementation.
+We will however make sure to never have multiple projects in planning phase at the same time, as this leads to cognitive overload while also delivering on an implementation for another projects.
```

Comment:
The goal ultimately is to avoid cognitive overload on individuals, so you can think of it as "not more than one project going on at the same time _per person_". It is very likely that as our team grows, we'll need to loosen this up to allow multiple groups of people working on different projects simultaneously. But at the same time, [we prefer to operate together as a team rather than a working group](https://about.sourcegraph.com/handbook/company/values#work-as-a-team) where everyone is working on different things. At least right now, most big projects we do will involve backend and frontend together, and we're just 2+2 engineers. We also have only one designer and one PM. So I believe at the moment if we do multiple large projects at the same time, it _will_ result in increased cognitive load for the team, which I want to avoid. And again, this may change in the future!